src/main/scala/io/citrine/lolo/bags/BaggedResult.scala

```diff
@@ -507,4 +697,29 @@ object BaggedResult {
     }
     scores.map(Math.max(floor, _))
   } ensuring (vec => vec.forall(_ >= 0.0))
+
+  /**
+    * Calculate the correlation coefficient given a sequence of covariance scores for each training point.
+    * In theory, the covariance is the sum of the individual covariance scores. But because each covariance score
+    * is noisy, the resulting sum can be larger in absolute value than `sigmaX * sigmaY`, which would translate into
+    * a correlation coefficient that is outside [-1.0, 1.0].
+    *
+    * @param covarianceScores sequence of Monte Carlo corrected covariance contributions for each training point
+    * @param sigmaX           uncertainty in first label
+    * @param sigmaY           uncertainty in second label
+    * @return                 correlation coefficient, guaranteed to be between -1.0 and 1.0
+    */
+  def rectifyCorrelationScores(covarianceScores: Vector[Double], sigmaX: Double, sigmaY: Double): Double = {
+    require(sigmaX >= 0.0 && sigmaY >= 0.0)
+    if (sigmaX == 0 || sigmaY == 0) return 0.0
+    val rho = covarianceScores.sum / (sigmaX * sigmaY)
+    // If the calculated rho is too large or too small, then the calculation is noisy and we set rho to 0.0
+    // TODO (PLA-8597): figure out a better way to rectify covariance noise
```

Comment:
I have thoughts on how to do this, but want to cover it in a separate PR in which I will benchmark the effect of the rectification strategy